cmake/toolchain/zephyr/0.11/target.cmake

```diff
@@ -16,8 +16,27 @@ set(CROSS_COMPILE_TARGET ${CROSS_COMPILE_TARGET_${ARCH}})
 set(SYSROOT_TARGET       ${CROSS_COMPILE_TARGET})
 
 if("${ARCH}" STREQUAL "xtensa")
-  set(SYSROOT_DIR ${TOOLCHAIN_HOME}/xtensa/${SOC_NAME}/${SYSROOT_TARGET})
-  set(CROSS_COMPILE ${TOOLCHAIN_HOME}/xtensa/${SOC_NAME}/${CROSS_COMPILE_TARGET}/bin/${CROSS_COMPILE_TARGET}-)
+  if("${SOC_SERIES}" STREQUAL "cavs_v15")
+    set(SYSROOT_DIR ${TOOLCHAIN_HOME}/xtensa/intel_apl_adsp/${SYSROOT_TARGET})
+    set(CROSS_COMPILE ${TOOLCHAIN_HOME}/xtensa/intel_apl_adsp/${CROSS_COMPILE_TARGET}/bin/${CROSS_COMPILE_TARGET}-)
+  elseif("${SOC_SERIES}" STREQUAL "cavs_v18")
+    set(SYSROOT_DIR ${TOOLCHAIN_HOME}/xtensa/intel_s1000/${SYSROOT_TARGET})
+    set(CROSS_COMPILE ${TOOLCHAIN_HOME}/xtensa/intel_s1000/${CROSS_COMPILE_TARGET}/bin/${CROSS_COMPILE_TARGET}-)
+  elseif("${SOC_SERIES}" STREQUAL "cavs_v20")
+    set(SYSROOT_DIR ${TOOLCHAIN_HOME}/xtensa/intel_s1000/${SYSROOT_TARGET})
+    set(CROSS_COMPILE ${TOOLCHAIN_HOME}/xtensa/intel_s1000/${CROSS_COMPILE_TARGET}/bin/${CROSS_COMPILE_TARGET}-)
+  elseif("${SOC_SERIES}" STREQUAL "cavs_v25")
+    set(SYSROOT_DIR ${TOOLCHAIN_HOME}/xtensa/intel_s1000/${SYSROOT_TARGET})
+    set(CROSS_COMPILE ${TOOLCHAIN_HOME}/xtensa/intel_s1000/${CROSS_COMPILE_TARGET}/bin/${CROSS_COMPILE_TARGET}-)
+  elseif("${SOC_SERIES}" STREQUAL "baytrail_adsp")
+    set(SYSROOT_DIR ${TOOLCHAIN_HOME}/xtensa/intel_byt_adsp/${SYSROOT_TARGET})
+    set(CROSS_COMPILE ${TOOLCHAIN_HOME}/xtensa/intel_byt_adsp/${CROSS_COMPILE_TARGET}/bin/${CROSS_COMPILE_TARGET}-)
+  elseif("${SOC_SERIES}" STREQUAL "broadwell_adsp")
+    set(SYSROOT_DIR ${TOOLCHAIN_HOME}/xtensa/intel_bdw_adsp/${SYSROOT_TARGET})
+    set(CROSS_COMPILE ${TOOLCHAIN_HOME}/xtensa/intel_bdw_adsp/${CROSS_COMPILE_TARGET}/bin/${CROSS_COMPILE_TARGET}-)
+  else()
+    message(FATAL_ERROR "Not compiler set for SOC_SERIES ${SOC_SERIES}")
+ endif()
```

Comment:
Because of:
https://github.com/zephyrproject-rtos/sdk-ng/pull/199

you should create a corresponding PR here:
https://github.com/zephyrproject-rtos/sdk-ng/blob/master/cmake/zephyr/target.cmake

Secondly, those are resulting in same `SYSROOT_DIR` and `CROSS_COMPILE`values, so could you cleanup the `elseif ... elseif ... elseif`.

Howabout cleaning this up completely, and do this instead:
```
<keep existing code above>
set(CROSS_COMPILE_TARGET  ${CROSS_COMPILE_TARGET_${ARCH}})
set(SYSROOT_TARGET        ${CROSS_COMPILE_TARGET})
set(ARCH_SOC              ${SYSROOT_TARGET})
set(ARCH_SOC_CROSS_COMPILE) # Only used for xtensa.

set(ARCH_SOC_xtensa_cavs_v15       "xtensa/intel_apl_adsp")
set(ARCH_SOC_xtensa_cavs_v18       "xtensa/intel_s1000")
set(ARCH_SOC_xtensa_cavs_v20       "xtensa/intel_s1000")
set(ARCH_SOC_xtensa_cavs_v25       "xtensa/intel_s1000")
set(ARCH_SOC_xtensa_baytrail_adsp  "xtensa/intel_byt_adsp")
set(ARCH_SOC_xtensa_broadwell_adsp "xtensa/intel_bdw_adsp")

if("${ARCH}" STREQUAL "xtensa")
  set(ARCH_SOC ${ARCH_SOC_xtensa_${SOC_SERIES}})
  set(ARCH_SOC_CROSS_COMPILE ${ARCH_SOC_xtensa_${SOC_SERIES}}/)
  if(NOT ARCH_SOC)
    message(FATAL_ERROR "Not compiler set for SOC_SERIES ${SOC_SERIES}")
  endif()
endif()

set(SYSROOT_DIR   ${TOOLCHAIN_HOME}/${ARCH_SOC}/${SYSROOT_TARGET})
set(CROSS_COMPILE ${TOOLCHAIN_HOME}/${ARCH_SOC_CROSS_COMPILE}${CROSS_COMPILE_TARGET}/bin/${CROSS_COMPILE_TARGET}-)

if("${ARCH}" STREQUAL "x86")
<keep existing code below>
```